attestation-service/src/token/ear_broker.rs

```diff
@@ -144,13 +144,17 @@ pub struct EarAttestationTokenBroker {
 }
 
 impl EarAttestationTokenBroker {
-    pub fn new(config: Configuration) -> Result<Self> {
-        let policy_engine = PolicyEngineType::OPA.to_policy_engine(
-            Path::new(&config.policy_dir),
-            include_str!("ear_default_policy_cpu.rego"),
-            "default_cpu.rego",
-        )?;
-        info!("Loading default AS policy \"default_cpu.rego\"");
+    pub async fn new(config: Configuration) -> Result<Self> {
+        let policy_engine =
+            PolicyEngineType::OPA.to_policy_engine(Path::new(&config.policy_dir))?;
+
+        policy_engine
+            .set_default_policy("default_cpu", include_str!("ear_default_policy_cpu.rego"))
+            .await?;
+
+        policy_engine
+            .set_default_policy("default_gpu", include_str!("ear_default_policy_gpu.rego"))
+            .await?;
```

Comment:
Oh.  Sorry for bothering, but the `set_default_policy` and `set_policy` are somehow overlapped for the only difference is whether override or not. Imo we can simply write file here
```rust
if !Path::new(<policy_dir>/default_cpu).exists() {
  fs::write(include_str!("ear_default_policy_gpu.rego"), <policy_dir>/default_cpu)?;
}
```

This could help avoid `async`-ify call stack, and also avoid a much similar member function.

Reply:
TBO, the `set_default_policy` interface is only called at creation time throughout the policy engine's lifecycle. Methods are usually a runtime conception, not initialization. Here, we want to determine the policy engine's default policy only during initialization. A good approach would be to create a builder form like

```rust
let policy_engine = PolicyEngineBuilder::new()
    .with_default_policy(<name>, <content>)
    .with_default_policy(<name>, <content>)
    .build();
```

This seems a bit heavy in some way, so I recommend simply determining the path directly above. Anyway, introducing a new function especially overlapped with an existing one is not ideal fwiw.

Reply:
@Xynnn007 The builder pattern seems like overkill for now. 

Remember that the OPA policy engine is just one implementation of the policy engine trait. The trait provides a `set_policy` method because the details of how the policies are stored are up to the implementation of the trait. Writing the policy directly to the filesystem in the caller breaks the interface of the policy engine. Setting a policy and setting a default policy seem to be two distinct operations that the policy engine trait should support so I like the idea of having `set_default_policy` be a simple wrapper.